code/parse/sexp.cpp

```diff
@@ -4361,7 +4367,7 @@ int get_sexp()
 			// was closing quote not found?
 			if (*(Mp + 1 + len) != '\"') {
 				error_display(0, "Unexpected end of quoted string embedded in sexp!");
-				skip_sexp();
+				skip_sexp(true);
```

Comment:
Logically-speaking, this should be `skip_sexp(false)`, as the closing quote not being found means that it will never turn off `within_quotes` and thus skip the entire rest of the file. Unless stopping all parsing of the file is intentional? In which case, I feel like calling `skip_sexp()` here at all gives the wrong impression.

Reply:
Skipping the rest of the file or string is indeed the intention.  Since the quote never ends, the parser should go as far as it can.  And since that is essentially skipping the rest of the sexp, `skip_sexp()` works here IMO.  The alternative is to use `Mp += (len + 1);` which admittedly is more efficient but I think is less clear.

Reply:
I think `skip_sexp()` is less clear because it's not immediately obvious that the entire rest of the file will be skipped.

Reply:
It's consistent with the pattern for every other error check in this section.  But I can add a comment; perhaps that will help.

Reply:
That works.